forge_tests/TestOperatorStore.sol

```diff
@@ -0,0 +1,158 @@
+        JBAccountingContextConfig[] memory _accountingContexts = new JBAccountingContextConfig[](1);
+        _accountingContexts[0] =
+            JBAccountingContextConfig({token: JBTokens.ETH, standard: JBTokenStandards.NATIVE});
+        _terminalConfigurations[0] =
+            JBTerminalConfig({terminal: _terminal, accountingContextConfigs: _accountingContexts});
+
+        _projectZero = _controller.launchProjectFor({
+            owner: makeAddr("8008"),
+            projectMetadata: _projectMetadata,
+            fundingCycleConfigurations: _cycleConfig,
+            terminalConfigurations: _terminalConfigurations,
+            memo: ""
+        });
+
+        _projectOne = _controller.launchProjectFor({
+            owner: _projectOwner,
+            projectMetadata: _projectMetadata,
+            fundingCycleConfigurations: _cycleConfig,
+            terminalConfigurations: _terminalConfigurations,
+            memo: ""
+        });
+    }
+
+    function testFailMostBasicAccess() public {
+        // Package up cycle config.
+        JBFundingCycleConfig[] memory _cycleConfig = new JBFundingCycleConfig[](1);
+        _cycleConfig[0].mustStartAtOrAfter = 0;
+        _cycleConfig[0].data = _data;
+        _cycleConfig[0].metadata = _metadata;
+        _cycleConfig[0].groupedSplits = new JBGroupedSplits[](0);
+        _cycleConfig[0].fundAccessConstraints = new JBFundAccessConstraints[](0);
+        
+        vm.prank(makeAddr("8008"));
+        _controller.reconfigureFundingCyclesOf(_projectOne, _cycleConfig, "");
+    }
+
+    function testFailSetOperators() public {
+        // Pack up our permission data
+        JBOperatorData[] memory opData = new JBOperatorData[](1);
+
+        uint256[] memory permIndexes = new uint256[](257);
+
+        // Push an index higher than 255
+        for (uint256 i; i < 257; i++) {
+            permIndexes[i] = i;
+
+            opData[0] = JBOperatorData({
+                operator: address(0),
+                domain: _projectOne,
+                permissionIndexes: permIndexes
+                });
+
+        // Set em.
+        vm.prank(_projectOwner);
+        _opStore.setOperatorOf(_projectOwner, opData[0]);
+        }
+    }
+
+    function testSetOperators() public {
+        // Pack up our permission data
+        JBOperatorData[] memory opData = new JBOperatorData[](1);
+
+        uint256[] memory permIndexes = new uint256[](256);
+
+        // Push an index higher than 255
+        for (uint256 i; i < 256; i++) {
+            permIndexes[i] = i;
+
+            opData[0] = JBOperatorData({
+                operator: address(0),
+                domain: _projectOne,
+                permissionIndexes: permIndexes
+                });
+
+        // Set em.
+        vm.prank(_projectOwner);
+        _opStore.setOperatorOf(_projectOwner, opData[0]);
+
+        // verify
+        bool _check = _opStore.hasPermission(address(0), _projectOwner, _projectOne, permIndexes[i]);
```

Comment:
nit> in another loop

Maybe it isn't super consistent in our current codebase (pleading guilty as charged), but we should (try to) use a formalised template, something like Setup/Test/Check comments to have clearer structure accross tests, as well as include natspecs 

```
// Setup: Create array to pass to foo() to do magic stuff
```
creating some arrays and stuff, calling some other fn (if integration) or setting mocks (if units)

```
// Test: Caling foo()
```
the tx we test now (or multiple tx)

```
// Check: Is result greater than Namibia GDP?
```
whatever assert/loop assert/etc (we group the assert in a loop even if test are calling in a loop)